src/plugins/internalnotification/internalnotification.c

```diff
@@ -0,0 +1,263 @@
+/**
+ * @file
+ *
+ * @brief Source for internalnotification plugin
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#include "internalnotification.h"
+
+#if PLUGIN_INTERNALNOTIFICATION_VERBOSE
+#include <stdio.h>
+#endif
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include <kdbhelper.h>
+
+struct _KeyRegistration
+{
+	char * name;
+	int * variable;
+	struct _KeyRegistration * next;
+};
+typedef struct _KeyRegistration KeyRegistration;
+
+struct _ListPointer
+{
+	KeyRegistration * head;
+	KeyRegistration * tail;
+};
+typedef struct _ListPointer ListPointer;
+
+
+static KeyRegistration * elektraInternalnotificationAddNewRegistration (ListPointer * listPointer)
+{
+	KeyRegistration * item = elektraMalloc (sizeof *item);
+	if (item == NULL)
+	{
+		return NULL;
+	}
+	item->next = NULL;
+
+	if (listPointer->head == NULL)
+	{
+		// Inizialize list
+		listPointer->head = listPointer->tail = item;
+	}
+	else
+	{
+		// Make new item end of list
+		listPointer->tail->next = item;
+		listPointer->tail = item;
+	}
+
+	return item;
+}
+
+static void elektraInternalnotificationUpdateRegisteredKeys (ListPointer * listPointer, KeySet * keySet)
```

Comment:
The parameter `listPointer` is not necessary you could call `elektraPluginGetData` in the function.